beacon-chain/p2p/gossipsub_peer_controller.go

```diff
@@ -0,0 +1,146 @@
+package p2p
+
+import (
+	"context"
+	"math"
+	"sync"
+	"time"
+
+	"github.com/OffchainLabs/prysm/v6/beacon-chain/p2p/gossipsubcrawler"
+	"github.com/OffchainLabs/prysm/v6/cmd/beacon-chain/flags"
+	"github.com/ethereum/go-ethereum/p2p/enode"
+	"github.com/pkg/errors"
+)
+
+const (
+	peerPerTopic = 20
```

Comment:
@kasey Let's think through this together. At what point should the controller stop dialing more peers proactively ?

Reply:
How `20` was chosen?

IMO, this number should depend of the topic family.
It's harder to find 20 peers for every data column sidecars topics than for, let's say, blocks.

Reply:
Yeah this will be replace by a per-topic-family policy for  (fanout + subscriptions) that we discussed @nalepae.